src/dsp_tools/utils/xmlupload/xmlupload.py

```diff
@@ -400,20 +400,29 @@ def _create_resource(
     res_type: type,
     resource: XMLResource,
     resource_iri: str | None,
-    bitstream_information: dict[str, Any] | None,
+    bitstream_information: BitstreamInfo | None,
     con: Connection,
     permissions_lookup: dict[str, Permissions],
     id2iri_mapping: dict[str, str],
 ) -> tuple[str, str] | None:
     properties = resource.get_propvals(id2iri_mapping, permissions_lookup)
     try:
+        if bitstream_information:
+            bitstream_info: dict[str, str | Permissions] | None = {
+                "value": bitstream_information.local_file,
+                "internal_file_name": bitstream_information.internal_file_name,
+            }
+            if bitstream_information.permissions and bitstream_info:
+                bitstream_info["permissions"] = bitstream_information.permissions
+        else:
+            bitstream_info = None
```

Comment:
Would it be too difficult to change the class `ResourceInstance`? It would be more natural to teach `ResourceInstance` how to cope with `BitstreamInfo`, if possible.

Reply:
You're right that this would be the clean implementation. But as this is just a preparatory step to get rid of ResourceInstance alsong with the ResourceInstanceFactory, I considered it the "way of least resistance" not to touch the internals of ResourceInstance. This conversion will of course be of a short life span, and the replacement of ResourceInstance will be able to handle BitstreamInfo directly.